### PR TITLE
Implement wasi dir path rename

### PR DESCRIPTION
**Pull request by bnjjj**
# Description
Related to #2557 

I also add command to wasi-test-generator to generate only for specific tests and not always for all tests.

### CODE REVIEW

**Review by syrusakbary — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -564,6 +564,9 @@ test-capi-integration-%:
 test-wasi-unit:
 	cargo test --manifest-path lib/wasi/Cargo.toml --release
 
+test-wasi:
+	cargo test --release --tests $(compiler_features) -- wasi::wasitests::snapshot1
```
Comment (Hywan) [inline Makefile:568]: Why specifically the `snapshot1`?

File: lib/c-api/wasmer.h [hunk 2]
```diff
@@ -62,9 +62,18 @@
...
+
+// The `compiler` feature has been enabled for this build.
+#define WASMER_COMPILER_ENABLED
+
 // The `wasi` feature has been enabled for this build.
 #define WASMER_WASI_ENABLED
 
+// The `middlewares` feature has been enabled for this build.
+#define WASMER_MIDDLEWARES_ENABLED
+
```
Comment (Hywan) [inline lib/c-api/wasmer.h:76]: Please, unstage this from the PR.

File: tests/wasi-wast/src/wasitests.rs [hunk 3]
```diff
@@ -147,7 +147,7 @@ fn compile_wasm_for_version(
             .create(true)
             .open(&temp_wasi_rs_file_name)
             .unwrap();
-        actual_file.write_all(b"#![feature(wasi_ext)]\n").unwrap();
+        // actual_file.write_all(b"#![feature(wasi_ext)]\n").unwrap();
```
Comment (Hywan) [inline tests/wasi-wast/src/wasitests.rs:150]: Please remove the commented line if not useful anymore.

**Comment by syrusakbary** [comment]
```graphql
{
  getPackage(name:"syrusakbary/python") {
    viewerHasRole(role:ADMIN)
  }
}
```

**Comment by bnjjj** [comment]
Any idea of what could be wrong here @syrusakbary  ?

**Comment by syrusakbary** [comment]
Tests were failing on wasmer `master`, now everything should be fine. I think triggering a merge again should solve it.

bors r+

**Comment by syrusakbary** [comment]
I think we hit a edge case @bnjjj 
```
---- wasi::wasitests::snapshot1::host_fs::fd_rename_path::cranelift::universal stdout ----
Running wasi wast `tests/wasi-wast/wasi/snapshot1/fd_rename_path.wast`
Error: failed to run WASI `_start` function: failed with stdout: ""
stderr: "thread 'main' panicked at 'cannot create the directory: Os { code: 20, kind: AlreadyExists, message: "File exists" }', /tmp/.tmp4mb8cH/wasi_modified_version_fd_rename_path.rs:10:35
```

**Comment by bnjjj** [comment]
Ok will try to reproduce and add more tests ASAP
